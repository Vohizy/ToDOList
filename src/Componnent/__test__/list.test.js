import { fireEvent, render, screen } from "@testing-library/react";
import  renderer from "react-test-renderer";
import List from "../list";



/* describe("list of input",() => {
    it("all list",() => {
        const tree = renderer.create(<List value={value} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
}) */
describe("test for the list component",()=>{
    const value = [
        "hello","world","!!!!"
    ]
    
    
    it('have any more in list',() => {
        
       const{getAllByTestId} = render(<List value={value} />)
        const selectElementToInteract = getAllByTestId("lists")
    
        expect(selectElementToInteract).toHaveLength(3)
    }) 
    it("input is exactily here",()=>{
        render(<List value={value}/>)
        const checks =screen.getByTestId("checke2")
        const selectElementToInteract = screen.getByRole('list')
        expect(selectElementToInteract).toBeInTheDocument()


        
    })
    it("there are in input type checkbox here",() => {
        render(<List value={value}/>)
        const getList = screen.getByTestId("checke2")
        expect(getList).toBeInTheDocument();
    })
     it("length of the array is decrement if i check",()=>{
       const mockCheck = jest.fn();
        render(<List value={value} f={mockCheck}/>)
        const checks = screen.getByTestId("checke1")
        fireEvent.click(checks)
        expect(mockCheck.mock.calls.length).toBe(1);


    })
    it('test list of snapshot',() => {
        const tree = renderer.create(<List value={value}/>).toJSON()
        expect(tree).toMatchSnapshot();
    }) 
})