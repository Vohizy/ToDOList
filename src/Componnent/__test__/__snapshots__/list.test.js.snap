// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for the list component test list of snapshot 1`] = `
<ul
  className="list"
>
  <li
    data-testid="lists"
  >
    <div
      className="li-div"
    >
      hello
      <input
        data-testid="checke1"
        onClick={[Function]}
        type="checkbox"
      />
    </div>
  </li>
  <li
    data-testid="lists"
  >
    <div
      className="li-div"
    >
      world
      <input
        data-testid="checke2"
        onClick={[Function]}
        type="checkbox"
      />
    </div>
  </li>
  <li
    data-testid="lists"
  >
    <div
      className="li-div"
    >
      !!!!
      <input
        data-testid="checke3"
        onClick={[Function]}
        type="checkbox"
      />
    </div>
  </li>
</ul>
`;
